crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -184,24 +186,29 @@ impl CheckpointHandler {
         let system_state_summary =
             get_sui_system_state(&checkpoint_object_store)?.into_sui_system_state_summary();
 
-        let epoch_event = transactions
+        let epoch_event_opt = transactions
             .iter()
-            .flat_map(|t| t.events.as_ref().map(|e| &e.data))
-            .flatten()
-            .find(|ev| ev.is_system_epoch_info_event())
-            .unwrap_or_else(|| {
-                panic!(
-                    "Can't find SystemEpochInfoEvent in epoch end checkpoint {}",
-                    checkpoint_summary.sequence_number()
-                )
-            });
-
-        let event = bcs::from_bytes::<SystemEpochInfoEvent>(&epoch_event.contents)?;
+            .find_map(|t| {
+                t.events.as_ref()?.data.iter().find_map(|ev| {
+                    if ev.is_system_epoch_info_event() {
+                        Some(bcs::from_bytes::<SystemEpochInfoEvent>(&ev.contents))
+                    } else {
+                        None
+                    }
+                })
+            })
+            .transpose()?;
+        if epoch_event_opt.is_none() {
+            warn!(
```

Comment:
I think you can still have an assert here to ensure that this is only None when we are in safe mode, which you can get from the system state.